internal/server/data/access_key.go

```diff
@@ -155,16 +156,9 @@ func ValidateAccessKey(db *gorm.DB, authnKey string) (*models.AccessKey, error)
 		}
 
 		t.ExtensionDeadline = time.Now().UTC().Add(t.Extension)
-		// if the context wasn't set already, set it from the key so we can save
-		org := OrgFromContext(db.Statement.Context)
-		if org == nil {
-			org, err = GetOrganization(db, ByID(t.OrganizationID))
-			if err != nil {
-				return nil, fmt.Errorf("loading organization: %w", err)
-			}
-			db.Statement.Context = WithOrg(db.Statement.Context, org)
-		}
-		if err := SaveAccessKey(db, t); err != nil {
+
+		tx = NewTransaction(tx.GormDB(), t.OrganizationID)
```

Comment:
The naming is maybe a bit misleading. `NewTransaction` doesn't create a new DB transaction, it uses an existing reference to an `sql.Tx`, but sets a new orgID for the transaction.

Reply:
This also took me a moment to realize what was going on, and how `Exec()` works inside the transaction.

Reply:
I just noticed this is done in the wrong place. I'll add a comment and move it to outside of this block.

Reply:
This `NewTransaction` wouldn't be necessary if we didn't have the `panic` thta was requested in these comments. It's kind of unfortunate that we're forcing unnecessary code because of safeguards that aren't correct in all cases.

Reply:
I was able to remove this by moving where we panic:

https://github.com/infrahq/infra/compare/dnephin/data-interface-org-id-panic?expand=1

I'll open that as a follow up